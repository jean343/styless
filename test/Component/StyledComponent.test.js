import React from 'react';
import renderer from 'react-test-renderer';
import styled, {css} from "styled-components";
import 'jest-styled-components';

test('Support the styled-components syntax', () => {
    const Div = styled.div`
		width: ${400}px;
		width: 400px;
		border: 1px solid #000;
		border: ${"1px"} ${"solid"} ${"#000"};
        background: ${props => props.primary ? 'palevioletred' : 'white'};
        color: ${props => props.primary ? 'white' : 'palevioletred'};
		color2: ${props => props.noChild && "#5cb85c"}; // @brand-success
		color3: @color;
		${props => props.disabled ? "" : "box-shadow: inset 0 0 0 30px rgba(255, 255, 255, 0.3)"};
	`;
    expect(renderer.create(<Div color="red" primary noChild/>).toJSON()).toMatchSnapshot();
});

test('Attaching additional props', () => {
    const Input = styled.input.attrs({
        // we can define static props
        type: 'password',

        // or we can define dynamic ones
        margin: props => props.size || '1em',
        padding: props => props.size || '1em'
    })`
        color: palevioletred;
        font-size: 1em;
        border: 2px solid palevioletred;
        border-radius: 3px;

        /* here we use the dynamically computed props */
        margin: ${props => props.margin};
        padding: ${props => props.padding};

        margin2: @margin;
        padding2: @padding;
`;
    expect(renderer.create(<div>
        <Input placeholder="A small text input" size="1em"/>
        <br/>
        <Input placeholder="A bigger text input" size="2em"/>
    </div>).toJSON()).toMatchSnapshot();
});

test('Support the styled-components css', () => {
    const Div = styled.div`
        padding: @padding;
        ${props => props.bordered && css`
          border: 1px solid @border;
        `}
	`;
    expect(renderer.create(<Div padding="1rem" border="red"/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div bordered padding="1rem" border="red"/>).toJSON()).toMatchSnapshot();
});

test('Support @media queries', () => {
    const Div = styled.div`
		@media screen and (min-width: 900px) {
            article {
              padding: @padding;
            }
        }
	`;
    expect(renderer.create(<Div padding="1rem" media="foo"/>).toJSON()).toMatchSnapshot();
});

test('Support &&&', () => {
    const Div = styled.div`
        &&& {
            color: palevioletred;
            font-weight: @bold;
        }
        &&&second {
            color: red;
        }
	`;
    expect(renderer.create(<Div/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div bold="bold"/>).toJSON()).toMatchSnapshot();
});

test('Support Element', () => {
    const Div = styled.div`
        a {
            color: red;
            &.is-focused:not(.is-open) > .Select-control {
                cursor: text;
            }
        }
	`;
    expect(renderer.create(<Div/>).toJSON()).toMatchSnapshot();
});

test('Support placeholder on the root', () => {
    const Div = styled.div`
        &::placeholder {
            color: HSL(216, 15%, 65%);
        }
	`;
    expect(renderer.create(<Div/>).toJSON()).toMatchSnapshot();
});

test('Support styling other components', () => {
    const Child1 = styled.div`
        color: brown;
	`;
    const Child2 = styled.div`
        color: red;
	`;
    const Div = styled.div`
        ${Child1},
        ${Child2}{
            width: 400px;
        }
	`;
    expect(renderer.create(<Div><Child1/><Child2/></Div>).toJSON()).toMatchSnapshot();
});

test('Support SC mixins', () => {
    const hover = css`
        &:hover {
            text-decoration: underline;
        }
    `;
    const Icon = styled.div`
	`;
    const Div = styled.div`
	    ${hover};
	    ${props => hover};
	    border-bottom: @color;
	    
		&:hover, &:hover ${Icon} {
			color: inherit;
		}
	`;
    expect(renderer.create(<span><Icon/><Div color="red"/></span>).toJSON()).toMatchSnapshot();
});