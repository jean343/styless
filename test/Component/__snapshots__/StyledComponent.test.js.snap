// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#20, box-shadow 1`] = `
<div
  className=""
/>
`;

exports[`#20, box-shadow 2`] = `
.c0 {
  box-shadow: inset 0 0 0 30px #ffffff80;
}

<div
  className="c0"
  disabled={true}
/>
`;

exports[`#20, not 1`] = `
.c0 {
  box-shadow: inset 0 0 0 30px red;
}

<div
  className="c0"
/>
`;

exports[`#20, not 2`] = `
<div
  className=""
  disabled={true}
/>
`;

exports[`Attaching additional props 1`] = `
.c0 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  margin: 1em;
  padding: 1em;
  margin2: 1em;
  padding2: 1em;
}

.c1 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  margin: 2em;
  padding: 2em;
  margin2: 2em;
  padding2: 2em;
}

<div>
  <input
    className="c0"
    margin="1em"
    padding="1em"
    placeholder="A small text input"
    size="1em"
    type="password"
  />
  <br />
  <input
    className="c1"
    margin="2em"
    padding="2em"
    placeholder="A bigger text input"
    size="2em"
    type="password"
  />
</div>
`;

exports[`Support &&& 1`] = `
.c0.c0.c0 {
  color: palevioletred;
}

.c0.c0.c0second {
  color: red;
}

<div
  className="c0"
/>
`;

exports[`Support &&& 2`] = `
.c0.c0.c0 {
  color: palevioletred;
  font-weight: bold;
}

.c0.c0.c0second {
  color: red;
}

<div
  className="c0"
/>
`;

exports[`Support @media queries 1`] = `
@media screen and (min-width:900px) {
  .c0 article {
    padding: 1rem;
  }
}

<div
  className="c0"
  media="foo"
/>
`;

exports[`Support Element 1`] = `
.c0 a {
  color: red;
}

.c0 a.is-focused:not(.is-open) > .Select-control {
  cursor: text;
}

<div
  className="c0"
/>
`;

exports[`Support SC mixins 1`] = `
.c1 {
  border-bottom: red;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1:hover,
.c1:hover.c0 {
  color: inherit;
}

<span>
  <div
    className="c0 "
  />
  <div
    className="c1"
    color="red"
  />
</span>
`;

exports[`Support placeholder on the root 1`] = `
.c0::-webkit-input-placeholder {
  color: #98a3b3ff;
}

.c0::-moz-placeholder {
  color: #98a3b3ff;
}

.c0:-ms-input-placeholder {
  color: #98a3b3ff;
}

.c0::placeholder {
  color: #98a3b3ff;
}

<div
  className="c0"
/>
`;

exports[`Support styling other components 1`] = `
.c2 {
  color: brown;
}

.c4 {
  color: red;
}

.c0 .c1,
.c0 .c3 {
  width: 400px;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  />
  <div
    className="c3 c4"
  />
</div>
`;

exports[`Support the styled-components css 1`] = `
.c0 {
  padding: 1rem;
}

<div
  className="c0"
/>
`;

exports[`Support the styled-components css 2`] = `
.c0 {
  padding: 1rem;
  border: 1px solid red;
}

<div
  className="c0"
/>
`;

exports[`Support the styled-components syntax 1`] = `
.c0 {
  width: 400px;
  width: 400px;
  border: 1px solid #000;
  border: 1px solid #000;
  background: palevioletred;
  color: white;
  color2: #5cb85c;
  color3: red;
  box-shadow: inset 0 0 0 30px rgba(255,255,255,0.3);
}

<div
  className="c0"
  color="red"
/>
`;
